File: project_euler/01/016.go
// Copyright [2017] <dmnsn7@gmail.com>

package main

import "fmt"

const cn = 1000

func multi2(s string) string {
	t := ""
	carry := 0
	for i := 0; i < len(s); i++ {
		digit := carry + int(s[i]-'0')*2
		t += string(digit%10 + '0')
		carry = digit / 10
	}

	if carry != 0 {
		t += string(carry + '0')
	}

	return t
}

func main() {
	num := "1"
	for i := 0; i < cn; i++ {
		num = multi2(num)
	}

	sum := 0
	for i := 0; i < len(num); i++ {
		sum += int(num[i] - '0')
	}

	fmt.Printf("%d\n", sum)
}

File: project_euler/01/016.py
#!/usr/bin/env python3
'''Copyright [2017] <dmnsn7@gmail.com>'''

N = 1000


def multi_2(num_s):
    num_t = ''
    carry = 0
    for _, cha in enumerate(num_s):
        digit = carry + int(cha) * 2
        num_t += str(digit % 10)
        carry = digit // 10

    if carry != 0:
        num_t += str(carry)

    return num_t


def main():
    num = '1'
    for _ in range(N):
        num = multi_2(num)

    add_up = 0
    for _, dig in enumerate(num):
        add_up += int(dig)

    print(add_up)


if __name__ == '__main__':
    main()
